source/collimation.f90

```diff
@@ -3300,20 +3055,20 @@ subroutine collimate_end_collimator()
 
         ! Indicate wether this is a secondary / tertiary / other particle;
         !  note that 'scatterhit' (equals 8 when set) is set in SCATTER.
-        if(db_name1(icoll)(1:3).eq.'TCP'   .or. &
-           db_name1(icoll)(1:4).eq.'COLM'  .or. &
-           db_name1(icoll)(1:5).eq.'COLH0' .or. &
-           db_name1(icoll)(1:5).eq.'COLV0'       ) then
+        if(cdb_cNameUC(icoll)(1:3).eq.'TCP'   .or. &
```

Comment:
Hmm. The primary/secondary/tertiary/other logic should probably in the future not be based on names... Maybe some flag in the collDB?

Reply:
Absolutely. This is very LHC-specific, and the new DB format is designed not to be. It will be addressed in the PR where I add the new DB format.